src/configure/configure.ts

```diff
@@ -408,98 +446,75 @@ class Orchestrator {
         if (this.inputs.sourceRepository.repositoryProvider === RepositoryProvider.Github) {
             repoAnalysisResult = await repoAnalysisHelper.getRepositoryAnalysis(this.inputs.sourceRepository, this.inputs.pipelineConfiguration.workingDirectory.split('\\').join('/'));
         }
-
         extensionVariables.templateServiceEnabled = false;
+        var appropriatePipelines: PipelineTemplate[] = [];
 
-        let appropriatePipelines;
-        // TO:DO- Get applicable pipelines for the repo type and azure target type if target already selected
-
-        if (extensionVariables.templateServiceEnabled) {
-            repoAnalysisResult = null;
-            appropriatePipelines = await vscode.window.withProgress(
+        if (!extensionVariables.templateServiceEnabled) {
+            var localPipelines = await vscode.window.withProgress(
                 { location: vscode.ProgressLocation.Notification, title: Messages.analyzingRepo },
-                () => templateHelper.analyzeRepoAndListAppropriatePipeline2(
-                    this.inputs.azureSession,
+                () => templateHelper.analyzeRepoAndListAppropriatePipeline(
                     this.inputs.sourceRepository.localPath,
                     this.inputs.sourceRepository.repositoryProvider,
                     repoAnalysisResult,
                     this.inputs.pipelineConfiguration.params[constants.TargetResource])
             );
-        }
-        else {
-            appropriatePipelines = await vscode.window.withProgress(
+            appropriatePipelines = localPipelines;
+        } else {
+            repoAnalysisResult = {
+                "applicationSettingsList": [
+                    {
+                        "language": "Docker",
+                        "buildTargetName": "Dockerfile",
+                        "deployTargetName": "Azure:AKS",
+                        "workingDirectory": "wddddddddddddd"
+                    }
+                ]
+            };
+            var remotePipelines: PipelineTemplate[] = await vscode.window.withProgress(
                 { location: vscode.ProgressLocation.Notification, title: Messages.analyzingRepo },
-                () => templateHelper.analyzeRepoAndListAppropriatePipeline(
+                () => templateHelper.analyzeRepoAndListAppropriatePipeline2(
+                    this.inputs.azureSession,
                     this.inputs.sourceRepository.localPath,
                     this.inputs.sourceRepository.repositoryProvider,
                     repoAnalysisResult,
                     this.inputs.pipelineConfiguration.params[constants.TargetResource])
             );
+            appropriatePipelines = remotePipelines;
         }
 
+        let pipelineMap: Map<string, PipelineTemplate[]> = new Map();
+
+        appropriatePipelines.forEach(element => {
+            if (pipelineMap.has(element.label)) {
+                pipelineMap.get(element.label).push(element);
```

Comment:
Is the intention here is to merge local and remote templates? Two questions:
1. Why do we need to do that? If remote template with label exist, should we not ignore local template?
2. Even if we need this, should we not do this inside analyzeRepoAndListAppropriatePipeline2() as analyzeRepoAndListAppropriatePipeline does not return both type of templates?

Reply:
The intention here is to have a list mapped by label, to have a dropdown with unique labels, in which the user will choose the template.
The local and remote filtering is done at a later stage, where the resource is selected.